lldb-eval/parser.cc

```diff
@@ -748,7 +739,7 @@ ExprResult Parser::Run(Error& error) {
   // Explicitly return ErrorNode if there was an error during the parsing. Some
   // routines raise an error, but don't change the return value (e.g. Expect).
   if (error) {
-    return std::make_unique<ErrorNode>();
+    return std::make_unique<ErrorNode>(ctx_->GetEmptyType());
```

Comment:
It would be possible to use an empty ptr here, but then we would need to make sure that we never invoke methods on the type. In any case, we should decide how we represent invalid/empty types, currently they are instances with IsValid() == false. We could change the convention to represent them with null, but I would really prefer to not mix this (i.e., allow bith null and Invalid type).

I think the current way (with all the GetEmptyType) is a bit safer against null pointer problems, but I agree it is more verbose. Do you think we should change this to use and check nulls consistently everywhere?